cloudapp/src/app/excel-export/excel-export-service.ts

```diff
@@ -0,0 +1,137 @@
+import { EventEmitter, Injectable } from '@angular/core'
+
+import { ColumnOption } from '../column-options'
+import { ColumnDefinition, COLUMNS_DEFINITIONS } from '../column-definitions'
+
+import * as XLSX from 'sheetjs-style'
+import * as FileSaver from 'file-saver'
+import { RequestedResource, RequestedResourcesService } from '../requested-resources-service/requested-resources-service'
+
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ExcelExportService {
+  private static readonly FILE_NAME: string = 'requested-resources'
+  private static readonly EXCEL_TYPE: string = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet;charset=UTF-8'
+  private static readonly EXCEL_EXTENSION: string = '.xlsx'
+
+  private static readonly PAGE_SIZE: number = 100
+
+  constructor(
+    private requestedResourcesService: RequestedResourcesService
+  ) { }
+
+
+  async generateExcel(circDeskCode: string, libraryCode: string, columnOptions: ColumnOption[]) {
+
+    const columnDefinitions: ColumnDefinition[] = this.getColumnDefinitions(columnOptions)
+
+    this.requestedResourcesService.findRequestedResources(
```

Comment:
Instead of using .then, await the call to findRequestedResources.

Reply:
Resolved with commit b075a3c